pytorch_blade/torch_blade/onnx_backends/backend_testbed.py

```diff
@@ -387,21 +410,26 @@ def _appendNode(self, old_node):
         self._current_segment.appendNode(node)
         for src_out, dst_out in zip(old_node.outputs(), node.outputs()):
             self._orig2segment_value_map[src_out] = dst_out
+            self._segment2orig_value_map[dst_out] = src_out
 
     def _segment_to_subgraph(self):
         # graph.copy() would change the value debugName
         subgraph = self._current_segment.copy()
-
-        input_list = subgraph.input_list()
-        src_input_list = self._current_segment.input_list()
+        # swap, need original values to look up self._segment2orig_value_map
+        subgraph, self._current_segment = self._current_segment, subgraph
 
         all_used_values = set()
         all_inter_outputs = set()
         for node in subgraph.nodes():
             for inp in node.inputs():
                 all_used_values.add(inp)
             for out in node.outputs():
-                all_inter_outputs.add(out)
+                if out in self._segment2orig_value_map:
+                    orig_out = self._segment2orig_value_map[out]
+                    if orig_out.uses():
+                        all_inter_outputs.add(out)
+                else:
+                    all_inter_outputs.add(out)
```

Comment:
Are these modified to filter out the unused value from per-original node subgraph outputs?

I think filtering out the subgraph outputs from `maybe_graph_output` at L434 would be more clear:
```python
subgraph_outputs = set(out for out in maybe_graph_output if out used in original graph)
```

Reply:
refactored.